### ISSUE TITLE
The page scrolls down too much on click

### ISSUE THREAD

**Issue by JaneJeon** [TryGhost/London#24]
Hi, so the expected behaviour when you click a post from the [front page](https://i.imgur.com/6gksaSG.jpg) is that the post [loads from the top](https://i.imgur.com/ijtCrDz.jpg)

However, this ONLY works when the site hasn't been scrolled. If the main page is [scrolled down even a little](https://i.imgur.com/TP3fu2h.jpg) when you click on a post, then [the post is loaded like halfway scrolled down already](https://i.imgur.com/BTYY6Bi.jpg).

I've been digging thru the source code and I think the issue has to do with the `swup` library, but I have no idea how I would fix that.

Any ideas?

**Comment by JaneJeon** [comment]
P.S. the 1.0.0 release works, so it's _definitely_ the swup library

**Comment by mindshaped** [comment]
@JaneJeon same problem here! meanwhile I applied a drty fix (applied via site footer code injection, code below) by scrolling to top on the swup contentReplaced callback.. but still, the original version and demo on https://london.ghost.io/ is way smoother cause it doesn't animate the whole scrolled vertical distance, looking forward to a proper fix :)

```html
<script>
window.onload = function(){  
      swup.on('contentReplaced', function(){
           var body = $("html, body");
           body.stop().animate({scrollTop:0}, 300);
      });
}
</script>
```

**Comment by JaneJeon** [comment]
@mindshaped thanks for the fix!

**Comment by jilljenn** [comment]
I still have the bug, even on 1.0.0. I'm trying to solve this problem.

**Comment by jilljenn** [comment]
I finally chose @mindshaped injection which is easy to put and good enough.

@JohnONolan Is the problem directly related to `initInfiniteScroll`?
Or just should there be some anchor to main content on [this line](https://github.com/TryGhost/London/blob/master/default.hbs#L58) of `default.hbs`, as suggested [there](https://swup.js.org/common-issues#scroll-control-offset)? Or just disable swup scrolling?

There seems to be a [swup scroll plugin](https://swup.js.org/plugins/scroll-plugin) that contains `doScrollingRightAway` for example.

I don't think swup is currently used the way it should be.

**Comment by ojhp** [comment]
I've managed to pull this change out of my own fork. I'm hoping this does what people want. When you go to a new page the scrolling plugin will scroll you back up to the top.

**Comment by minimaluminium** [comment]
Hey everyone! The issue was fixed in 8948faa. The fix uses a [suggestion](https://github.com/swup/swup/issues/213#issuecomment-593128654) from the creator of swup instead of using the scroll plugin as the theme doesn't use the plugin elsewhere.
